src/api.ts

```diff
@@ -230,6 +231,20 @@ export const resolvers = {
     }
   },
   Query: {
+    async testPerforceConnection(parent: any, args: { connectionSettings: Settings }): Promise<OperationStatus> {
+      try {
+        const isSuccess = changePerforceManagerSingleton({
+          connectionString: args.connectionSettings.PerforceConnectionString,
+          timeout: parseInt(args.connectionSettings.PerforceTimeout || "5000", 10),
+          username: args.connectionSettings.PerforceUser
+        });
+        return { isSuccess, reason: 'make sure your configuration is correct' }
+      } catch (e) {
+        getLogger("Perforce").error("Failed to connect to Perforce", { e, settings: args.connectionSettings });
+        console.error(`Failed to init perforce manager with port: ${args.connectionSettings}`);
+        return { isSuccess: false, reason: e?.toString() };
```

Comment:
Maybe add a fallback reason?

Reply:
WDYM?

Reply:
If e is undefined we will get no reason for the error, so maybe there should be a default reason there